web/modules/custom/joinup_news/config/install/joinup_news.settings.yml

```diff
@@ -0,0 +1,110 @@
+# Format of array:
+# @code
+# [
+#   <parent moderation> => [
+#     <target state> => [
+#       <source state> => [
+#         <role1>
+#         <role2>
+#         .
+#         .
+#       ]
+#     ]
+#   ]
+# ]
+# @endcode
+#
+# The array is a four dimensions array. The first level contains the
+# value of the moderation of the parent entity. The second level is the
+# target state which is tha state that the entity transits to. Each target
+# state is an array of allowed source states which in every check, is the
+# current state of the entity. Finally, the source states are arrays
+# of roles that are allowed to perform this action.
+#
+# The reverse indexing here (source states
+# indexed by the target state) is to follow the state_machines module logic
+# of indexing states.
+transitions:
+#moderation: 0|post-moderated.
+  0:
+    draft:
+      __new__:
+        - moderator
+        - rdf_entity-collection-member
+    proposed:
+      proposed:
+        - moderator
+        - rdf_entity-collection-facilitator
+      validated:
+        - moderator
+        - rdf_entity-collection-facilitator
+      in_assessment:
+        - moderator
+        - rdf_entity-collection-facilitator
+    validated:
+      __new__:
+        - moderator
+        - rdf_entity-collection-member
+        - rdf_entity-collection-facilitator
+      draft:
+        - moderator
+        - rdf_entity-collection-member
+        - rdf_entity-collection-facilitator
+      proposed:
+        - moderator
+        - rdf_entity-collection-facilitator
+      validated:
+        - moderator
+        - rdf_entity-collection-facilitator
+    in_assessment:
+      validated:
+        - moderator
+        - rdf_entity-collection-member
+    deletion_request: { }
+#moderation: 1|pre-moderated.
+  1:
+    draft:
+      __new__:
+        - moderator
+        - rdf_entity-collection-member
+    proposed:
+      __new__:
+        - moderator
+        - rdf_entity-collection-member
+      draft:
+        - moderator
+        - rdf_entity-collection-member
+      proposed:
+        - moderator
+        - rdf_entity-collection-facilitator
+      validated:
+        - moderator
+        - rdf_entity-collection-facilitator
+      in_assessment:
+        - moderator
+        - rdf_entity-collection-facilitator
+    validated:
+      __new__:
+        - moderator
+        - rdf_entity-collection-facilitator
+      draft:
+        - moderator
+        - rdf_entity-collection-facilitator
+      proposed:
+        - moderator
+        - rdf_entity-collection-facilitator
+      validated:
+        - moderator
+        - rdf_entity-collection-facilitator
+      deletion_request:
+        - moderator
+        - rdf_entity-collection-facilitator
+    in_assessment:
+      validated:
+        - moderator
+        - rdf_entity-collection-member
+    deletion_request:
+      validated:
+        - moderator
+        - rdf_entity-collection-administrator
```

Comment:
Coding standards do not check for presence of empty line at the end of the yaml file, but it's better to have it (like core modules do).